### PR TITLE
auto generate session id

### PR DESCRIPTION
**Pull request by andreashille**
* either take from url or if none valid is present generate a new one

Signed-off-by: Andreas Hille <andreas.hille@tngtech.com>

### CODE REVIEW

File: frontend/src/Components/LoginPage.tsx [hunk 1]
```diff
@@ -79,7 +80,13 @@ const styling = css`
 const ProtoLoginPage = ({ socket }: { socket: WebSocketApi }) => {
   const firstInputRef: React.RefObject<HTMLInputElement> = React.useRef(null);
   const [user, setUser] = React.useState('');
-  const [session, setSession] = React.useState('');
+  const path = window.location.pathname.slice(1);
+  const sessionId = path.match(
```
Comment (siskavera) [inline frontend/src/Components/LoginPage.tsx:84]: This generates a new ID whenever anything changes, including typing in the username field

File: template.yaml [hunk 2]
```diff
@@ -21,6 +21,11 @@ Resources:
       WebsiteConfiguration:
         IndexDocument: index.html
         ErrorDocument: error.html
+        RoutingRules:
+          - RedirectRule:
```
Comment (siskavera) [inline template.yaml:25]: Out of curiousity: was this needed for the case when the user writes something random as sessionId?

File: frontend/src/Components/LoginPage.tsx [hunk 3]
```diff
@@ -76,10 +77,22 @@ const styling = css`
   }
 `;
 
+function randomString(length: number) {
```
Comment (siskavera) [inline frontend/src/Components/LoginPage.tsx:80]: Nice manual sessionID generator :D

File: frontend/src/Components/LoginPage.tsx (hunk 1 above)
Comment (andreashille) [reply frontend/src/Components/LoginPage.tsx]: fixed

File: frontend/src/Components/App.test.tsx [hunk 4]
```diff
@@ -29,7 +29,8 @@ describe('The App component', () => {
     expect(container).toHaveTextContent('Connecting...');
   });
 
-  it('creates a socket connection and displays the login window', () => {
+  it('creates a socket connection and displays the login window with an autogemerated session link', () => {
```
[the remark this answers is not in the document]
Comment (andreashille) [reply frontend/src/Components/App.test.tsx:32]: fixed

File: template.yaml (hunk 2 above)
Comment (andreashille) [reply template.yaml:25]: No, it should redirect 404 ot the index html

File: frontend/src/Components/LoginPage.tsx (hunk 3 above)
Comment (andreashille) [reply frontend/src/Components/LoginPage.tsx:82]: shorter than uuid ;-)

**Pull request merged**